halo2_proofs/src/plonk.rs

```diff
@@ -120,9 +120,21 @@ where
                 "unexpected version byte",
             ));
         }
+        // Maximum allowed value for parameter `k`, the log-size of the circuit.
+        const MAX_CIRCUIT_SIZE: u32 = 32;
+
         let mut k = [0u8; 4];
         reader.read_exact(&mut k)?;
-        let k = u32::from_le_bytes(k);
+        let k = u32::from_be_bytes(k);
+        if k > MAX_CIRCUIT_SIZE {
```

Comment:
u32 seems like a bad choice of a type then no? I think `u8` should do the job (I'm not expecting to see any 2^256-two adic curve anytime soon LOL)